mrmustard/lab_dev/circuit_components.py

```diff
@@ -248,21 +268,17 @@ def on(self, modes: Sequence[int]) -> CircuitComponent:
                 msg = f"Expected ``{len(modes)}`` modes, found ``{len(subset.modes)}``."
                 raise ValueError(msg)
 
-        wires = Wires(
+        ret = self.light_copy()
+        ret._wires = Wires(
             modes_out_bra=modes if ob else set(),
```

Comment:
I understand this to be: if ob is not defined, make modes_out_bra equal to set(). But why wouldn't ob already be equal to set()?

Reply:
Yes, and `bool(set())` evaluates to `False`. So if `ob` is `set()`, we make `modes_out_bra = set()`.